core/src/View/MDSpan/Kokkos_MDSpan_Layout.hpp

```diff
@@ -216,6 +234,85 @@ KOKKOS_INLINE_FUNCTION auto mapping_from_view_mapping(const VM &view_mapping) {
 #endif
 }
 
+template <unsigned ScalarSize>
+struct Padding {
+  static constexpr unsigned div =
+      ScalarSize == 0 ? 0 : MEMORY_ALIGNMENT / ScalarSize;
+  static constexpr unsigned mod =
+      ScalarSize == 0 ? 0 : Kokkos::Impl::MEMORY_ALIGNMENT % ScalarSize;
+
+  // If memory alignment is a multiple of the trivial scalar size then attempt
+  // to align.
+  static constexpr unsigned align  = ScalarSize != 0 && mod == 0 ? div : 0;
+  static constexpr unsigned div_ok = (div != 0) ? div : 1;
+
+  KOKKOS_INLINE_FUNCTION
+  static constexpr size_t stride(size_t const N) {
+    return ((align != 0) &&
+            ((static_cast<int>(Kokkos::Impl::MEMORY_ALIGNMENT_THRESHOLD) *
+              static_cast<int>(align)) < N) &&
+            ((N % div_ok) != 0))
+               ? N + align - (N % div_ok)
+               : N;
+  }
+};
```

Comment:
This is duped from ViewMapping we need to only have one version I think. We could change View Mapping to use this as an alias or something